examples/ui-demo/src/components/small-cards/SolanaCard.tsx

```diff
@@ -0,0 +1,167 @@
+import { queryClient } from "@/app/config";
+import { useToast } from "@/hooks/useToast";
+import { useSigner, useSignerStatus } from "@account-kit/react";
+import { LAMPORTS_PER_SOL } from "@solana/web3.js";
+import { useMutation, useQuery } from "@tanstack/react-query";
+import { useMemo } from "react";
+import { toBytes, toHex } from "viem";
+import { LoadingIcon } from "../icons/loading";
+import { UserAddressTooltip } from "../user-connection-avatar/UserAddressLink";
+import { Button } from "./Button";
+import { createTransfer } from "./solana/createSolanaTransfer";
+import * as solanaNetwork from "./solana/solanaNetwork";
+
+export const SolanaCard = () => {
+  const signer = useSigner();
+  const status = useSignerStatus();
+  const solanaSigner = useMemo(() => {
+    if (!signer) return;
+    if (!status.isConnected) return;
+    return signer.experimental_toSolanaSigner();
+  }, [signer, status.isConnected]);
+  const connection = solanaNetwork.connect();
+  const { setToast } = useToast();
+
+  const { data: balance = 0, isLoading: isBalanceLoading } = useQuery({
+    queryKey: ["solanaBalance", solanaSigner?.address],
+    queryFn: async () => {
+      if (!solanaSigner) return 0;
+      return (
+        (await solanaNetwork.balance(connection, solanaSigner!.address)) /
+        LAMPORTS_PER_SOL
+      );
+    },
+  });
+
+  const {
+    mutate,
+    isPending,
+    data: txHash,
+  } = useMutation({
+    mutationFn: async () => {
+      if (!solanaSigner) return;
+
+      const tx = await createTransfer({
+        fromAddress: solanaSigner.address!,
+        amount: 1000000,
+        toAddress: "tkhqC9QX2gkqJtUFk2QKhBmQfFyyqZXSpr73VFRi35C",
+        version: "legacy",
+        connection,
+      });
+
+      await solanaSigner.addSignature(tx);
+      return solanaNetwork.broadcast(connection, tx);
+    },
+    onSuccess: () => {
+      queryClient.invalidateQueries({
```

Comment:
nit: this seems to just result in refetching the old balance. i assume b/c the mutation is not waiting for enough confirmations?

Reply:
oh yea could be 🤔  lemme see if I can fix this

Reply:
nm figured it out, it had to do with the commitment level set on the connection. changing it to `confirmed` now matches the behaviour for when the tx comes back as well